packages/ra-core/src/i18n/useTranslateLabel.ts

```diff
@@ -30,13 +33,14 @@ export const useTranslateLabel = () => {
             return translate(
                 ...getFieldLabelTranslationArgs({
                     label: label as string,
+                    labelFromSourceContext: sourceContext?.getLabel(source),
```

Comment:
The issue is that I would have to also pass the unmodified source. You can see I pass the `finalSource` at L40. Besides, as the `labelFromSourceContext` would be `undefined` if `sourceContext` isn't present, I fail to see what it would change

Reply:
I disagree, `sourceContext?.getLabel(source)` will return `undefined` if there is no `sourceContext` but it will throw an error if `sourceContext.getLabel` is `undefined`

Reply:
I don't understand why you need to pass both labels. It shouldn't matter to `getFieldLabelTranslationArgs` where the label comes from.

Reply:
Because in the first label case, we don't want the inferred label fallback (users might pass something which is not a translation key) whereas in the second label case (the one from SourceContext) we do want to fallback to the inferred label